diff --git a/src/include/OpenImageIO/span_util.h b/src/include/OpenImageIO/span_util.h
index 89063e7f9..9c9524301 100644
--- a/src/include/OpenImageIO/span_util.h
+++ b/src/include/OpenImageIO/span_util.h
@@ -112,8 +112,8 @@ size_t
 spancpy(span<T> dst, size_t dstoffset, cspan<T> src, size_t srcoffset, size_t n)
 {
     // Where do the requests end (limited by span boundaries)?
-    size_t dstend = std::min(dstoffset + n, std::size(dst));
-    size_t srcend = std::min(srcoffset + n, std::size(src));
+    size_t dstend = std::min(dstoffset + n, std::size_t(std::size(dst)));
+    size_t srcend = std::min(srcoffset + n, std::size_t(std::size(src)));
     // How many can/should we copy?
     size_t ndst = dstend - dstoffset;
     size_t nsrc = srcend - srcoffset;
